### ISSUE TITLE
Relative path to CSS is invalid

### ISSUE THREAD

**Issue by dnagir** [staticmatic/staticmatic#10]
Path to CSS file is generated from the site root. The images and javascript files are generated in relation to the page.
So that:
    = stylesheets
    = img 'test.gif'
    = javascripts :site

Will give us:
    <link href="/stylesheets/screen.css" media="all" rel="stylesheet" type="text/css"/>
    <img alt="Test" src="images/test.gif"/>
    <script language="javascript" src="javascripts/site.js" type="text/javascript"></script>

The issues are:
1. The CSS will not be applied when opening file from the file system (path is invalid).
2. `AssetsHelper.img` docs specifies that the path is generated including the leading '/' which it doesn't (correctly!)

I guess that following things should be done:
1. Generate the CSS' src in relation to the current page but not the site root.
2. Update the `AssetsHelper.img` docs to remove the trailing '/' in the sample.

Using `--pre`

**Comment by STRd6** [comment]
I had the same issue but then learned about running the preview server rather than loading from the local file system.

**Comment by silvenon** [comment]
Yeah, that's been bugging me too, I'm glad I'm not crazy.

For now I guess we should just keep removing the slash, but I hope it'll be fixed soon.

**Comment by gardiner** [comment]
Just ran into the same thing. The assets helper looks as if he tries very hard to determine the relative path to the stylesheet, but somehow we still end up with the absolute path ('/stylesheets/bla....'). This isn't just a show-stopper for local preview, but prevents the generated site from being deployed into a subdirectory. Looking forward to seeing this fixed.

**Comment by gardiner** [comment]
There is a workaround which seems to work unless you use subdirectories for your stylesheets: In `lib/staticmatic/helpers/assets_helper.rb` you can replace line 53 with the following line:
    options[:href] = "#{relative_path}stylesheets/#{filename_without_extension}.css"

(This is actually just copy/paste/adjust of the respective line from the javascript helper.)

**Comment by silvenon** [comment]
It also works to type:
    %link{:type => 'text/css', :rel => 'stylesheet', :src => 'your_relative_path_here'}
instead of:
    = stylesheets

It works both for testing with WEBrick and for building the site.

**Comment by gardiner** [comment]
Good Point. Actually you could just write `<link href="stylesheets/foobar.css" rel="stylesheet" type="text/css"/>`, but it wouldn't be much fun any more ;)
It's still good to know that there are some ways around this "inconvenience".
